ramp4-pcar4/storylines-editor
# Autolegend

### Related Item(s)
[config-editor#34](https://github.com/ramp4-pcar4/config-editor/issues/34)
[config-editor#35](https://github.com/ramp4-pcar4/config-editor/issues/35)

### Changes
- [FEATURE] In the layers section of the config editor, there is now a button called "autopopulate legend". When you click this, a layer item will be created in the legend for each layer that does not already have one. Additionally, every layer item in the legend config that does not have a corresponding layer in the layers config will be removed. This acts as a temporary autolegend until one exists in RAMP.
- [FEATURE] Editing the layer's ID in the layers config automatically updates the `layerId` field for the layer item in the legend config, if it exists.

### Notes
* This PR also refactors a significant portion of the config editor code. Would be good to do a quick sanity check of previous functionality.

### Testing
Steps:
1. Open the config editor.
2. Add a number of layers.
3. Click autopopulate.
4. Ensure the legend config updates.
5. Remove a number of layers.
6. Click autopopulate again.
7. Ensure the legend config updates.


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ramp4-pcar4/storylines-editor/376)
